### PR TITLE
MF-1088 - Remove message payload content type

### LINKED ISSUE #1088: Remove `/ct` content-type topic extension
**FEATURE REQUEST**

1. Is there an open issue addressing this request? If it does, please add a "+1" reaction to the
   existing issue, otherwise proceed to step 2.

2. Describe the feature you are requesting, as well as the possible use case(s) for it.
Currently MQTT has a topic extension of `/ct`, which signifies content-type. This was needed because WS adapter had to know in advance which type of socket needs to be opened, and then influenced API of all other adapters.

It would be better approach to offload this content-type awareness to an application, and make the infrastructure (Mainflux) content-type agnostic.

After https://github.com/mainflux/mainflux/issues/1087 this will be possible and should be implemented.

3. Indicate the importance of this feature to you (must-have, should-have, nice-to-have).
Must-have

### PR DESCRIPTION
**Pull request by manuio**
Signed-off-by: Manuel Imperiale <manuel.imperiale@gmail.com>

Resolves #1088

### CODE REVIEW

**Review by drasko — APPROVED** [review]
This looks good, but it will be mandatory that @dusanb94 and @nmarcetic review as well.

Docs will need update, as this breaks the API.

**Review by drasko — DISMISSED** [review]
LGTM

File: transformers/senml/transformer.go [hunk 1]
```diff
@@ -20,20 +27,19 @@ var formats = map[string]senml.Format{
...
 // New returns transformer service implementation for SenML messages.
-func New() transformers.Transformer {
-	return transformer{}
+func New(contentType string) transformers.Transformer {
+	return transformer{
+		format: formats[contentType],
+	}
 }
 
 func (n transformer) Transform(msg broker.Message) (interface{}, error) {
```
Comment (dborovcanin) [inline transformers/senml/transformer.go]: You can rename `func (n transformer) Transform...` to `func (t transformer) Transform...`. Weird receiver name `n` is a legacy from `Normalizer`.

File: transformers/senml/transformer.go [hunk 2]
```diff
@@ -20,20 +27,19 @@ var formats = map[string]senml.Format{
...
+	format senml.Format
+}
 
 // New returns transformer service implementation for SenML messages.
-func New() transformers.Transformer {
-	return transformer{}
+func New(contentType string) transformers.Transformer {
+	return transformer{
+		format: formats[contentType],
+	}
```
Comment (dborovcanin) [inline transformers/senml/transformer.go]: It makes sense to have a default format (also, please rename `contentType` to `contentFormat`):
```go
format, ok := formats[contentFormat]
if !ok {
    format = formats[JSON]
}

return transformer {
    format: format,
} 
```

**Review by dborovcanin — DISMISSED** [review]

**Review by nmarcetic — DISMISSED** [review]

**Review by drasko — APPROVED** [review]
LGTM

**Review by mteodor — APPROVED** [review]

**Pull request merged**